junit-jupiter-api/src/main/java/org/junit/jupiter/api/condition/DisabledOnJreCondition.java

```diff
@@ -32,9 +32,17 @@ class DisabledOnJreCondition extends BooleanExecutionCondition<DisabledOnJre> {
 
 	@Override
 	boolean isEnabled(DisabledOnJre annotation) {
-		JRE[] versions = annotation.value();
-		Preconditions.condition(versions.length > 0, "You must declare at least one JRE in @DisabledOnJre");
-		return Arrays.stream(versions).noneMatch(JRE::isCurrentVersion);
+		JRE[] jres = annotation.value();
+		int[] featureVersions = annotation.versions();
+		Preconditions.condition(jres.length > 0 || featureVersions.length > 0,
+			"You must declare at least one JRE or version in @DisabledOnJre");
+		return Arrays.stream(jres).map(this::validateJre).noneMatch(JRE::isCurrentVersion)
+				&& Arrays.stream(featureVersions).noneMatch(JRE::isCurrentVersion);
```

Comment:
Could this be simplified by mapping `JRE`s to their version and joining the two streams?

Reply:
Indeed, I considered that.

We can certainly concatenate two streams and process them as one.

I'm not sure if it will perform any better, but it might make it a bit more readable. So I'll give it a try.

Reply:
Take a look at commit https://github.com/junit-team/junit5/pull/3931/commits/e675dd6c5272635ecba0bb1e9ebe85a82bbcb337.

It of course works, but I'm not sure it's any better in terms of readability.

Reply:
Thanks, I think it's better since it doesn't repeat the condition. :+1: